### ISSUE TITLE
Namespace conflicts in method completion

### ISSUE THREAD

**Issue by castwide** [castwide/solargraph#108]
Given the following code:

```ruby
class Foo
  def root_method; end
end

module Other
  class Foo
    def other_method; end
  end

  # @type [Foo]
  var1 = get_other_foo
  # @type [::Foo]
  var2 = get_foo
end
```

Since `var1` is declared inside `Other`, its type should be resolved as the nearest match, `Other::Foo`. Since `var2`'s type has a namespace prefix, its type should be the root `Foo` class. In v0.28.4, this works correctly when inferring the variable's type (e.g., in a hover), but requesting completions for `var2` erroneously returns `Other::Foo` instance methods.

This bug is fixed in the master branch and is scheduled for inclusion in v0.29.0.

**Comment by gencer** [comment]
@castwide, on `master` I still have this issue.

**Comment by castwide** [comment]
@gencer Do you have an example? When I test the above code, it correctly gives me `Other::Foo` methods for `var1` and `Foo` methods for `var2`.

![image](https://user-images.githubusercontent.com/3704922/49166231-eb6ae180-f300-11e8-9de5-b9d390cf0a41.png)

![image](https://user-images.githubusercontent.com/3704922/49166282-050c2900-f301-11e8-87a0-a00d202899cf.png)

**Comment by gencer** [comment]
@castwide, I've sent you an another project. This time I included multiple files from my project. Please have a look at your email and hope you can reproduce at on your end. 

BTW, When I try your example, yes, it just works. However, withing my existing project it doesn't. Email sent.

Hi Fred,

Instructions:

1.	Open project and go to “app/box/v1/entities.rb”
2.	Find line 198. You will see that i assigned comment with an object.
3.	Enter there and write “comment.” You wont see my models/classess.

**Comment by castwide** [comment]
@gencer This case might be stretching Solargraph's limits. Neither the `Comment` class in app/models/comment.rb nor its superclass `Sequel::Model` have methods that can be detected through static analysis.

I hope to improve support for Rails conventions in particular--e.g., recognizing that `has_many :users` exposes a `users` method--but it has a long way to go.

**Comment by castwide** [comment]
Ah! Thanks for the followup. Glad you caught it.

**Comment by castwide** [comment]
Fix released in v0.29.0.
